internal/dwarf/frame/table.go

```diff
@@ -0,0 +1,482 @@
+}
+
+func restoreextended(frame *FrameContext) {
+	reg, _ := util.DecodeULEB128(frame.buf)
+
+	oldrule, ok := frame.initialRegs[reg]
+	if ok {
+		frame.Regs[reg] = DWRule{Offset: oldrule.Offset, Rule: RuleOffset}
+	} else {
+		frame.Regs[reg] = DWRule{Rule: RuleUndefined}
+	}
+}
+
+func defcfa(frame *FrameContext) {
+	reg, _ := util.DecodeULEB128(frame.buf)
+	offset, _ := util.DecodeULEB128(frame.buf)
+
+	frame.CFA.Rule = RuleCFA
+	frame.CFA.Reg = reg
+	frame.CFA.Offset = int64(offset)
+}
+
+func defcfaregister(frame *FrameContext) {
+	reg, _ := util.DecodeULEB128(frame.buf)
+	frame.CFA.Reg = reg
+}
+
+func defcfaoffset(frame *FrameContext) {
+	offset, _ := util.DecodeULEB128(frame.buf)
+	frame.CFA.Offset = int64(offset)
+}
+
+func defcfasf(frame *FrameContext) {
+	reg, _ := util.DecodeULEB128(frame.buf)
+	offset, _ := util.DecodeSLEB128(frame.buf)
+
+	frame.CFA.Rule = RuleCFA
+	frame.CFA.Reg = reg
+	frame.CFA.Offset = offset * frame.dataAlignment
+}
+
+func defcfaoffsetsf(frame *FrameContext) {
+	offset, _ := util.DecodeSLEB128(frame.buf)
+	offset *= frame.dataAlignment
+	frame.CFA.Offset = offset
+}
+
+func defcfaexpression(frame *FrameContext) {
+	var (
+		l, _ = util.DecodeULEB128(frame.buf)
+		expr = frame.buf.Next(int(l))
+	)
+
+	frame.CFA.Expression = expr
+	frame.CFA.Rule = RuleExpression
+}
+
+func expression(frame *FrameContext) {
+	var (
+		reg, _ = util.DecodeULEB128(frame.buf)
+		l, _   = util.DecodeULEB128(frame.buf)
+		expr   = frame.buf.Next(int(l))
+	)
+
+	frame.Regs[reg] = DWRule{Rule: RuleExpression, Expression: expr}
+}
+
+func offsetextendedsf(frame *FrameContext) {
+	var (
+		reg, _    = util.DecodeULEB128(frame.buf)
+		offset, _ = util.DecodeSLEB128(frame.buf)
+	)
+
+	frame.Regs[reg] = DWRule{Offset: offset * frame.dataAlignment, Rule: RuleOffset}
+}
+
+func valoffset(frame *FrameContext) {
+	var (
+		reg, _    = util.DecodeULEB128(frame.buf)
+		offset, _ = util.DecodeULEB128(frame.buf)
+	)
+
+	frame.Regs[reg] = DWRule{Offset: int64(offset), Rule: RuleValOffset}
+}
+
+func valoffsetsf(frame *FrameContext) {
+	var (
+		reg, _    = util.DecodeULEB128(frame.buf)
+		offset, _ = util.DecodeSLEB128(frame.buf)
+	)
+
+	frame.Regs[reg] = DWRule{Offset: offset * frame.dataAlignment, Rule: RuleValOffset}
+}
+
+func valexpression(frame *FrameContext) {
+	var (
+		reg, _ = util.DecodeULEB128(frame.buf)
+		l, _   = util.DecodeULEB128(frame.buf)
+		expr   = frame.buf.Next(int(l))
+	)
+
+	frame.Regs[reg] = DWRule{Rule: RuleValExpression, Expression: expr}
+}
+
+func louser(frame *FrameContext) {
+	frame.buf.Next(1)
+}
+
+func hiuser(frame *FrameContext) {
+	frame.buf.Next(1)
+}
+
+func gnuargsize(frame *FrameContext) {
+	// The DW_CFA_GNU_args_size instruction takes an unsigned LEB128 operand representing an argument size.
+	// Just read and do nothing.
+	// TODO(kakkoyun): Implement this.
+	_, _ = util.DecodeSLEB128(frame.buf)
+}
+
+// TODO(kakkoyun): ? How to move cursor without corrupting?  Do we actually need to do this?
```

Comment:
cc @v-thakkar This was a futile attempt to skip diving into the actual implementation of the instructions. Better to remove this.